metal/pwm.h

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__PWM_H
+#define METAL__PWM_H
+
+/*! @brief Enums for PWM running modes. */
+typedef enum {
+    METAL_PWM_CONTINUOUS = 0,
+    METAL_PWM_ONE_SHOT = 1
+} metal_pwm_run_mode_t;
+
+/*! @brief Enums for Phase correct PWM. */
+typedef enum {
+    METAL_PWM_PHASE_CORRECT_DISABLE = 0,
+    METAL_PWM_PHASE_CORRECT_ENABLE = 1,
+} metal_pwm_phase_correct_t;
+
+/*! @brief Enums for Interrupts enable/disable. */
+typedef enum {
+    METAL_PWM_INTERRUPT_DISABLE = 0,
+    METAL_PWM_INTERRUPT_ENABLE = 1,
+} metal_pwm_interrupt_t;
```

Comment:
Not sure to put yet, but having a common METAL_ENABLE make more sense, instead every driver defining its own.
This can be done later.

Reply:
A set of common APIs / Enums / Macros for metal drivers can be defined for a more uniform implementation of all drivers. Not sure now much flexibility we have now, to re-define existing metal APIs.